Libplanet.Net.Tests/Consensus/ContextTest.cs

```diff
@@ -867,6 +869,603 @@ public async void PreCommitBlockToEndCommit()
+
+            context.HandleMessage(
+                new ConsensusCommit(
+                    TestUtils.CreateVote(privateKeys[0], 0, 1, 1, block.Hash, VoteFlag.Commit))
+                {
+                    Remote = new Peer(privateKeys[0].PublicKey),
+                });
+
+            context.HandleMessage(
+                new ConsensusCommit(
+                    TestUtils.CreateVote(privateKeys[1], 1, 1, 1, block.Hash, VoteFlag.Commit))
+                {
+                    Remote = new Peer(privateKeys[1].PublicKey),
+                });
+            context.HandleMessage(
+                new ConsensusCommit(
+                    TestUtils.CreateVote(privateKeys[2], 2, 1, 1, block.Hash, VoteFlag.Commit))
+                {
+                    Remote = new Peer(privateKeys[2].PublicKey),
+                });
+
+            var roundVoteSet = context.VoteSet(1);
+            Assert.True(roundVoteSet.HasTwoThirdCommit());
+            Assert.Equal(Step.EndCommit, context.Step);
+            Assert.Equal(1, context.Height);
+            Assert.Equal(1, context.Round);
+
+            Assert.Equal(VoteFlag.Commit, roundVoteSet.Votes[0].Flag);
+            Assert.Equal(VoteFlag.Commit, roundVoteSet.Votes[1].Flag);
+            Assert.Equal(VoteFlag.Commit, roundVoteSet.Votes[2].Flag);
+            Assert.Equal(VoteFlag.Null, roundVoteSet.Votes[3].Flag);
+            Assert.Equal(block.Hash, roundVoteSet.Votes[0].BlockHash);
+            Assert.Equal(block.Hash, roundVoteSet.Votes[1].BlockHash);
+            Assert.Equal(block.Hash, roundVoteSet.Votes[2].BlockHash);
+            Assert.Equal(block.Hash, roundVoteSet.Votes[3].BlockHash);
+        }
+
+        [Fact(Timeout = Timeout)]
+        public async void RoundSkipProposeToPreVoteBlockToPreCommit()
+        {
+            var (validators, privateKeys) = GetRandomValidators();
+
+            using ITransport transport =
+                TestUtils.CreateNetMQTransport(privateKeys[3], port: Port + 14);
+            var codec = new Codec();
+            BlockChain<DumbAction> blockChain =
+                TestUtils.CreateDummyBlockChain((MemoryStoreFixture)_fx);
+            var voteSent = false;
+            var messageReceived = new ManualResetEventSlim();
+
+            void IsPreCommitSent(ConsensusMessage message)
+            {
+                voteSent = message is ConsensusCommit;
+                if (voteSent)
+                {
+                    messageReceived.Set();
+                }
+            }
+
+            var consensusContext =
+                TestUtils.CreateStandaloneConsensusContext(
+                    blockChain,
+                    transport,
+                    newHeightDelay,
+                    3,
+                    blockChain.Tip.Index + 1,
+                    port: Port + 14,
+                    privateKey: privateKeys[3],
+                    validators: validators,
+                    watchConsensusMessage: IsPreCommitSent);
```

Comment:
IMO this is repeated in another test case. How about make this logic into a new method?